src/storage/src/hummock/sstable_store.rs

```diff
@@ -254,6 +260,52 @@ impl SstableStore {
             .map_err(HummockError::object_io_error)
     }
 
+    pub async fn preload_blocks(
+        &self,
+        sst: &Sstable,
+        mut start_index: usize,
+        mut end_index: usize,
+    ) -> HummockResult<Option<BatchBlockStream>> {
+        let object_id = sst.id;
+        while start_index < end_index
+            && self.block_cache.exists_block(object_id, start_index as u64)
+        {
+            start_index += 1;
+        }
+        if start_index >= end_index {
+            return Ok(None);
+        }
+        let start_offset = sst.meta.block_metas[start_index].offset as usize;
+        let mut end_offset = start_offset;
+        for idx in start_index..end_index {
+            if self.block_cache.exists_block(object_id, idx as u64) {
+                end_index = idx;
+                break;
+            }
+            end_offset += sst.meta.block_metas[idx].len as usize;
+        }
```

Comment:
Since we haven't pin block cache here, we can't guarantee that these blocks will still exist during later reading. Maybe we should preload block data between `start_index` and `end_index` without relying on the block cache, otherwise, for OLAP batch query with a small block cache size, streaming read will degrade into read.